Packs/SophosXGFirewall/Integrations/sophos_firewall_test.py

```diff
@@ -0,0 +1,1117 @@
+    assert result.outputs_prefix == 'SophosFirewall.User'
+    assert result.outputs.get('Name') == 'unitest3'
+
+
+def test_sophos_firewall_user_delete_command(requests_mock):
+    """
+    Scenario: Delete an existing user.
+    Given:
+     - User has provided valid credentials.
+    When:
+     - sophos_firewall_user_delete is called.
+    Then:
+     - Ensure outputs prefix is correct.
+     - Ensure a sample value from the API matches what is generated in the context.
+    """
+    from sophos_firewall import Client, sophos_firewall_user_delete_command
+    mock_response = load_mock_response('user_set.xml')
+    requests_mock.post(REQUEST_URL, text=mock_response)
+    client = Client(base_url=BASE_URL, verify=False, auth=('uname', 'passwd'), proxy=False)
+    result = sophos_firewall_user_delete_command(client, 'forunitest')
+    assert result.outputs_prefix == 'SophosFirewall.User'
+    assert result.outputs.get('Name') == 'forunitest'
+
+
+def test_prepare_builder_params(requests_mock):
+    """
+    Scenario: Prepare list objects for builder
+    Given:
+     - User has provided valid credentials.
+    When:
+     - Before a builder is used.
+    Then:
+     - Ensure the correct item is returned from the function based on the arguemnts.
+    """
+    from sophos_firewall import Client, prepare_builder_params
+    mock_response = load_mock_response('rule_get.xml')
+    requests_mock.get(REQUEST_URL, text=mock_response)
+    client = Client(base_url=BASE_URL, verify=False, auth=('uname', 'passwd'), proxy=False)
+    result = prepare_builder_params(client, {'members': ['Identity', 'Member']}, True,
+                                    'rule', 'SecurityPolicy', {'members': 'new'})
+    assert result == {'members': ['new']}
+
+
+def test_update_dict_from_params_using_path():
+    """
+    Scenario: Update dictionary from parameters using path.
+    Given:
+     - User has provided valid credentials.
+    When:
+     - A value is supposed to be added to a dictionary.
+    Then:
+     - Ensure the correct item is returned from the function based on the arguemnts.
+    """
+    from sophos_firewall import update_dict_from_params_using_path
+    result = update_dict_from_params_using_path({'a': ['b', 'c']}, {'a': ['b', 'c']}, {'a': ['d']})
+    assert result == {'a': ['d'], 'b': {'c': ['b', 'c']}}
+
+
+def test_check_error_on_response():
+    """
+    Scenario: Check for any errors in a response.
+    Given:
+     - User has provided valid credentials.
+    When:
+     - A response is returned from the API
+    Then:
+     - Ensure the correct item is returned from the function based on the arguemnts.
+    """
+    from sophos_firewall import check_error_on_response
+    good_result = json.loads(load_mock_response('rule_get.json'))
+    check_error_on_response(good_result)
+    bad_result = {'Status': 'No. of records Zero.'}
+    with pytest.raises(Exception):
+        check_error_on_response(bad_result)
```

Comment:
best to split to 2 test cases.
the first with good_result is suffixed `_happy`, the second as is.